### PR TITLE
Add test for 'Replace' function for PostgreSQL (and fix its type)

### PR DESCRIPTION
**Pull request by rpiotrow**
Addresses #247

 * **fix**: literals rendering 
 * **test**: more tests for select queries in PostgreSQL
 * **fix**: 'Replace' function type
 * **test**: add test for 'Replace' function for PostgreSQL
 * **fix**: update test schema

### CODE REVIEW

File: core/jvm/src/main/scala/zio/sql/expr.scala [hunk 1]
```diff
@@ -108,7 +108,21 @@ trait ExprModule extends NewtypesModule with FeaturesModule with OpsModule {
...
+        case TypeTag.TDouble    => Expr.Literal(a)
+        case TypeTag.TFloat     => Expr.Literal(a)
+        case TypeTag.TInt       => Expr.Literal(a)
+        case TypeTag.TLong      => Expr.Literal(a)
+        case TypeTag.TShort     => Expr.Literal(a)
+        case _                  => literalAsString(a)
+      }
+
+    private def literalAsString[A](a: A) =
+      Expr.Literal(s"'${a.toString}'").asInstanceOf[Expr[Features.Literal, Any, A]]
```
Comment (jczuchnowski) [inline core/jvm/src/main/scala/zio/sql/expr.scala:125]: Generally it's a good direction, but I have the same doubts about it as in https://github.com/zio/zio-sql/pull/269.
This really fits better into rendering.

Comment (jczuchnowski) [reply core/jvm/src/main/scala/zio/sql/expr.scala:125]: So TBH I would drop this change and make the test wrap the String params in single quotes for the moment. Ugly I know. But it's not a precedent. Someone will fix this in another PR in the renderer.

**Comment by jczuchnowski** [comment]
Some coordination concern, but otherwise awesome work that was much needed.

File: core/jvm/src/main/scala/zio/sql/expr.scala (hunk 1 above)
Comment (rpiotrow) [reply core/jvm/src/main/scala/zio/sql/expr.scala:125]: I can change it in the afternoon (together with conflict resolving), but I will need to add some nasty workaround to tests added to `PostgresModuleTest.scala`.

**Comment by rpiotrow** [comment]
Closed in favor of #319

**Pull request closed without merging**